moveit_core/kinematic_constraints/src/utils.cpp

```diff
@@ -273,3 +274,298 @@ moveit_msgs::Constraints kinematic_constraints::constructGoalConstraints(const s
+  {
+    if (it->first == "type")
+      continue;
+    else if (it->first == "joint_name")
+      constraint.joint_name = static_cast<std::string>(it->second);
+    else if (it->first == "weight")
+      constraint.weight = parseDouble(it->second);
+    else if (it->first == "position")
+    {
+      constraint.position = parseDouble(it->second);
+    }
+    else if (it->first == "tolerance")
+    {
+      constraint.tolerance_below = parseDouble(it->second);
+      constraint.tolerance_above = parseDouble(it->second);
+    }
+    else if (it->first == "tolerances")
+    {
+      if (!isArray(it->second, 2, it->first, "lower/upper tolerances"))
+        return false;
+
+      constraint.tolerance_below = parseDouble(it->second[0]);
+      constraint.tolerance_above = parseDouble(it->second[1]);
+    }
+    else if (it->first == "bounds")
+    {
+      if (!isArray(it->second, 2, it->first, "lower/upper bound"))
+        return false;
+
+      const double lower_bound = parseDouble(it->second[0]);
+      const double upper_bound = parseDouble(it->second[1]);
+
+      constraint.position = (lower_bound + upper_bound) / 2;
+      constraint.tolerance_below = constraint.position - lower_bound;
+      constraint.tolerance_above = upper_bound - constraint.position;
+    }
+    else
+    {
+      ROS_WARN_STREAM_NAMED(LOGNAME, "joint constraint contains unknown entity '" << it->first << "'");
+    }
+  }
+  return true;
+}
+
+static bool constructConstraint(XmlRpc::XmlRpcValue& params, moveit_msgs::PositionConstraint& constraint)
+{
+  for (XmlRpc::XmlRpcValue::iterator it = params.begin(); it != params.end(); ++it)
+  {
+    if (it->first == "type")
+      continue;
+    else if (it->first == "frame_id")
+      constraint.header.frame_id = static_cast<std::string>(it->second);
+    else if (it->first == "weight")
+      constraint.weight = parseDouble(it->second);
+    else if (it->first == "link_name")
+      constraint.link_name = static_cast<std::string>(it->second);
+    else if (it->first == "target_offset")
+    {
+      if (!isArray(it->second, 3, it->first, "x/y/z position"))
+        return false;
+
+      constraint.target_point_offset.x = parseDouble(it->second[0]);
+      constraint.target_point_offset.y = parseDouble(it->second[1]);
+      constraint.target_point_offset.z = parseDouble(it->second[2]);
+    }
+    else if (it->first == "region")
+    {
+      if (!isStruct(it->second, { "x", "y", "z" }, "region"))
+        return false;
+      constraint.constraint_region.primitive_poses.emplace_back();
+      constraint.constraint_region.primitives.emplace_back();
+      geometry_msgs::Pose& region_pose = constraint.constraint_region.primitive_poses.back();
+      shape_msgs::SolidPrimitive& region_primitive = constraint.constraint_region.primitives.back();
+
+      std::function<double(XmlRpc::XmlRpcValue&)> center = [](XmlRpc::XmlRpcValue& it) {
+        return parseDouble(it[0]) + (parseDouble(it[1]) - parseDouble(it[0])) / 2;
```

Comment:
This is hard to read as well. The center of the interval is just `(parseDouble(it[0]) + parseDouble(it[1])) / 2;`